File: libs/fst/CoreCrypto_DEPRECATED/CoreRandom.ml
(* -------------------------------------------------------------------- *)
let rng = Random.self_init ()

(* -------------------------------------------------------------------- *)
let random (i : int) =
  if i < 0 then
    invalid_arg "length must be non-negative";

  let s = String.create i in

  for i = 0 to i-1 do
    s.[i] <- char_of_int (Random.int 256)
  done;

  Bytes.abytes s

File: libs/fs/CoreCrypto/CoreRandom.fs
(* Copyright (C) 2012--2015 Microsoft Research and INRIA *)

module CoreRandom

open Org.BouncyCastle.Security

let provider = new SecureRandom()

let random (i : int) =
    if i < 0 then
        invalidArg "length" "must be non-negative";

    let bytes = Array.create i 0uy in
        lock provider (fun () -> provider.NextBytes(bytes, 0, i));
        Bytes.abytes bytes
